src/components/Accordion/PAccordion.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="p-accordion">
+    <template v-for="section in sections" :key="section">
+      <div class="p-accordion__section">
+        <div class="p-accordion__header">
+          <slot :name="`${section}-heading`">
+            <slot name="heading" :section="section">
+              {{ section }}
+            </slot>
+          </slot>
+          <p-button
+            size="sm"
+            icon="ChevronDownIcon"
+            flat
+            rounded
+            class="p-accordion__button"
+            :class="getButtonClass(String(section))"
+            @click="toggleSection(String(section))"
+          />
+        </div>
+
+        <PAutoHeightTransition>
+          <template v-if="isOpen(String(section))">
+            <div class="p-accordion__content">
+              <!-- this is needed because PAutoHeightTransition does not animate an elements padding correctly -->
+              <div class="p-accordion__content-padding">
+                <slot :name="`${section}-content`">
+                  <slot name="content" :section="section" />
+                </slot>
+              </div>
+            </div>
+          </template>
+        </PAutoHeightTransition>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup generic="const T extends Readonly<string[]>">
+  import { Ref, ref } from 'vue'
+  import { PAutoHeightTransition } from '@/components/AutoHeightTransition'
+
+  const props = defineProps<{
+    sections: T,
+  }>()
+
+  const open: Ref<T[number] | null> = ref(props.sections[0])
```

Comment:
Would like this to be configurable instead of always grabbing the first one - the type for this is also slightly incorrect since an empty array fulfills `string[]` and so will return `undefined` in this instance.

Reply:
Updated this to use a v-model. By default no sections are open.